LeanOA/ForMathlib/Analysis/CStarAlgebra/SpecialFunctions/Abs.lean

```diff
@@ -0,0 +1,172 @@
+/-
+Copyright (c) 2024 Jon Bannon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Bannon, Jireh Loreaux
+-/
+
+import LeanOA.ForMathlib.Analysis.SpecialFunctions.ContinuousFunctionalCalculus.Abs
+import LeanOA.ForMathlib.Analysis.SpecialFunctions.ContinuousFunctionalCalculus.PosPart
+import Mathlib.Analysis.CStarAlgebra.ContinuousFunctionalCalculus.Instances
+import Mathlib.Analysis.CStarAlgebra.ContinuousFunctionalCalculus.Order
+
+/-!
+# Absolute value of an operator defined via the continuous functional calculus
+
+This file provides basic API for `CFC.abs` for `CStarAlgebra`s.
+
+-/
+
+open scoped NNReal
+
+namespace CFC
+
+section NonUnital
+
+variable {A : Type*} [NonUnitalNormedRing A] [StarRing A]
+  [Module ℝ A] [SMulCommClass ℝ A A] [IsScalarTower ℝ A A]
+  [NonUnitalContinuousFunctionalCalculus ℝ (IsSelfAdjoint : A → Prop)]
+
+lemma mul_self_eq_mul_self {a : A} (ha : IsSelfAdjoint a) : a * a =
+    cfcₙ (fun (x : ℝ) ↦ x * x) a := by
+  conv_lhs => rw [← cfcₙ_id' ℝ a, ← cfcₙ_mul ..]
+
+variable [UniqueNonUnitalContinuousFunctionalCalculus ℝ A]
+
+lemma cfcₙ_sqrt_mul_self {a : A} (ha : IsSelfAdjoint a) :
+    cfcₙ Real.sqrt (a * a) = cfcₙ (fun x ↦ √(x * x)) a := by
+  rw [mul_self_eq_mul_self ha, ← cfcₙ_comp a (f := fun x ↦ x * x) (g := fun x ↦ √x),
+     Function.comp_def]
+
+variable {A : Type*} [PartialOrder A] [NonUnitalNormedRing A] [StarRing A]
+   [Module ℝ A] [SMulCommClass ℝ A A] [IsScalarTower ℝ A A]
+   [NonUnitalContinuousFunctionalCalculus ℝ (IsSelfAdjoint : A → Prop)]
+   [UniqueNonUnitalContinuousFunctionalCalculus ℝ≥0 A]
+   [StarOrderedRing A] [NonnegSpectrumClass ℝ A]
+
+lemma sqrt_eq_cfcₙ_real_sqrt {a : A} (ha : 0 ≤ a := by cfc_tac) :
+    CFC.sqrt a = cfcₙ Real.sqrt a := by
+  rw [sqrt_eq_iff _ (hb := cfcₙ_nonneg (A := A) (fun x _ ↦ Real.sqrt_nonneg x)),
+    ← cfcₙ_mul ..]
+  conv_rhs => rw [← cfcₙ_id (R := ℝ) a]
+  refine cfcₙ_congr fun x hx ↦ ?_
+  refine Real.mul_self_sqrt ?_
+  exact quasispectrum_nonneg_of_nonneg a ha x hx
+
+variable {A : Type*} [NonUnitalCStarAlgebra A]
```

Comment:
I think this is too strong for a lot of what follows.

Reply:
OK.